### ISSUE TITLE
Xiaomi 3rd party light "Dancelight" Support request !

### ISSUE THREAD

**Issue by kuperss** [rytilahti/python-miio#793]
Before submitting a new request, use the search to see if there is an existing issue for the device.

**Device information:**

  - Name(s) of the device: [Dancelight I Series Bulb]&[Dancelight I Series Downlight]
  - Link: http://www.dancelight.com.tw/index.php?option=module&lang=cht&task=showlist&id=577&index=3
             http://www.dancelight.com.tw/index.php?option=productm&lang=cht&task=showlist&id=1051&index=0

  - Model: [tsd.light.tsl001]
  - Hardware version: 
  - Firmware version: 2.0.3_0601
 
**Additional context**
If you know already about potential commands or any other useful information to add support for the device, please add that information here.

**Following are my ha configuration:**
light:
  - platform: xiaomi_miio
    name: dancelight
    host: 10.70.1.XX
    token: 4f80446e93f87d03524d2XXXXXXX

homeassistant:
  customize:
    light.dancelight:
      friendly_name: 小光頭
      homebridge_name: 小光頭

**Following are Home Assistant Error Log:**
Logger: homeassistant.components.xiaomi_miio.light
Source: components/xiaomi_miio/light.py:194
Integration: Xiaomi Miio (documentation, issues)
First occurred: 上午11:54:44 (1 occurrences)
Last logged: 上午11:54:44

Unsupported device found! Please create an issue at https://github.com/syssi/philipslight/issues and provide the following data: tsd.light.tsl001

 **Mi Home app logs about these device:**
"線上裝置" means "device online"
SmartHome 51602:[DEBUG]-08-08 01:36:51.317 processResult in result={"code":0,"message":"ok","result":{"list":[{"did":"264215579","token":"4f80446e93f87d03524d2bb2bXXXXXXXX","longitude":"0.00000000","latitude":"0.00000000","name":"小光頭","pid":"0","localip":"10.70.1.XXX","mac":"50:EC:50:3E:XXXXX","ssid":"KupeR's HomE","bssid":"30:23:03:CC:96:90","parent_id":"","parent_model":"","show_mode":1,"**model":"tsd.light.tsl001"**,"adminFlag":1,"shareFlag":0,"permitLevel":16,"isOnline":true,"desc":"線上裝置 ","extra":{"isSetPincode":0,"fw_version":"2.0.3_0601","needVerifyCode":0,"isPasswordEncrypt":0},"uid":169359XXXX,"pd_id":66694,"password":"","p2p_id":"","rssi":-58,"family_id":0,"reset_flag":0},{"did":"28605XXXX","token":"69427a47635a4550466a757052XXXXXX","longitude":"0.00000000","latitude":"0.00000000","name":"小米AI音箱","pid":"0","localip":"10.70.1.XXX","mac":"E0:B6:55:D2:XXXXX","ssid":"KupeR's HomE","bssid":"30:23:03:96:CE:6E","parent_id":"","parent_model":"","show_mode":1,"model":"xiaomi.wifispeaker.s12","adminFlag":1,"shareFlag":0,"permitLevel":16,"isOnline":false,"desc":"裝置離線 ","extra":{"isSetPincode":0,"fw_version":"1.58.4","needVerifyCode":0,"isPasswordEncrypt":0},"uid":169359XXXX,"pd_id":464,"password":"","p2p_id":"","rssi":-63,"family_id":0,"reset_flag":0}],"virtualModels":[{"model":"zhimi.airpurifier.m1","state":0,"url":""},{"model":"yunmi.waterpurifier.v2","state":0,"url":""},{"model":"zimi.powerstrip.v2","state":0,"url":""}]}}

PS: DANCELIGHT is a very popular brand in Taiwan. Recently three items have been added to the xiaomi mija app. Please provide miio support, thank you very much

**Comment by rezmus** [comment]
mitospec.

**Comment by rytilahti** [comment]
@rezmus good point! Yes, I think it would make sense to consolidate standard actions, where available. All miot improvements are pending on that API improvement PR, alas, I haven't had time to dig into that and I'm not sure when I'll find some time for that in the near future.
